oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ScopeAuthorizationManagerFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.core;
+
+import java.util.Arrays;
+
+import org.springframework.security.authorization.AuthorityAuthorizationManager;
+
+/**
+ * @author Mario Petrovski
+ */
+public final class ScopeAuthorizationManagerFactory {
```

Comment:
Spring Security tends to name factories by pluralizing what is being created. Since this can create `AuthorizationManager`s, will you please call this `OAuth2AuthorizationManagers` instead?